src/pal/src/locale/utf8.cpp

```diff
@@ -1463,7 +1463,7 @@ class UTF8Encoding
                 }
 
                 // get pSrc 2-byte aligned
-                if (((int)pSrc & 0x1) != 0) {
+                if (((unsigned long)pSrc & 0x1) != 0) {
```

Comment:
This would cast to 64 bits on 32 bit architectures. Can you please use `size_t` instead.